swift-generator/src/main/java/com/facebook/swift/generator/template/StructContext.java

```diff
@@ -51,6 +52,26 @@ public boolean getHasUniqueFieldTypes()
       return fieldTypesSet.size() == fieldTypesList.size();
     }
 
+    public boolean getHasUniqueFieldTypesUnderErasure()
+    {
+        List<String> fieldTypesList = Lists.transform(this.fields, new Function<FieldContext, String>() {
+            public String apply(FieldContext field) {
+                // remove generics
+                String javaType = field.getJavaType();
+                int startTypeParameter = javaType.indexOf('<');
```

Comment:
See https://docs.oracle.com/javase/7/docs/api/java/lang/Character.html#isJavaIdentifierStart(char) and https://docs.oracle.com/javase/7/docs/api/java/lang/Character.html#isJavaIdentifierPart(char). Tl;dr: no :)